pubsub/streaming-analytics/build.gradle

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id 'java'
+}
+
+repositories {
+    mavenCentral()
+    maven {
+        url = uri('https://repository.apache.org/content/repositories/snapshots/')
+    }
+
+    maven {
+        url = uri('https://packages.confluent.io/maven/')
+    }
+
+    maven {
+        url = uri('https://repo.maven.apache.org/maven2/')
+    }
+}
+
+dependencies {
+    implementation 'com.github.spotbugs:spotbugs-annotations:4.5.0'
+    implementation 'org.apache.beam:beam-sdks-java-core:2.34.0'
+    implementation 'org.apache.beam:beam-sdks-java-io-google-cloud-platform:2.34.0'
+    implementation 'org.apache.beam:beam-examples-java:2.34.0'
+    implementation 'org.slf4j:slf4j-api:1.7.32'
+    implementation 'org.slf4j:slf4j-jdk14:1.7.32'
+    implementation 'com.github.spotbugs:spotbugs-annotations:4.5.0'
+    runtimeOnly 'org.apache.beam:beam-runners-direct-java:2.34.0'
+    runtimeOnly 'org.apache.beam:beam-runners-google-cloud-dataflow-java:2.34.0'
+    testImplementation 'com.google.cloud:google-cloud-core:2.3.2'
+}
+
+group = 'com.example'
+version = '1.0.0-SNAPSHOT'
+description = 'pubsub-streaming'
+java.sourceCompatibility = JavaVersion.VERSION_1_8
```

Comment:
Java 8 will stop its active support in 3 months. Unfortunately Beam doesn't support Java 17 yet, so can we use Java 11?

Does it work if we don't pin to a specific Java version? If so, let's not pin to a Java version.

https://endoflife.date/java